### ISSUE TITLE
`redis_exporter` fails to connect to Redis when password is provided via `redis_password_file` or environment variable

### ISSUE THREAD

**Issue by jesseops** [grafana/agent#2435]
I can successfully authenticate to redis when putting the password directly in the configuration file, but if I use an environment variable or password file, I get only `Couldn't connect to redis instance`.

I enabled verbose redis logging to check that the connection is initiated:

```
1:M 26 Oct 2022 15:50:56.320 - Accepted 172.18.0.1:59848
1:M 26 Oct 2022 15:50:56.321 - Client closed connection
```
this repeats every 60s and coincides with the message logged by grafana-agent:
```
Oct 26 15:50:56 grafanapoc grafana-agent[3448008]: time="2022-10-26T15:50:56Z" level=error msg="Couldn't connect to redis instance"
```

I have not been able to find any way to increase the redis_exporter log level to get more details. The password file is at least _potentially_ being read, as I can change the configuration to reference a non-existent file and observe that the integration fails to load.

Examples:

## Does work

```file=/etc/grafana-agent.yaml
...
redis_exporter:
    enabled: true
    relabel_configs:
      - replacement: <hostname>
        target_label: instance
    redis_addr: "redis://localhost:6379"
    redis_password: "<password>"
...
```

## Does not work

Example config using environment variable, added `-config.expand-env` to `/etc/default/grafana-agent`:`CUSTOM_ARGS` and appended `REDIS_PASSWORD=<password>` to `/etc/default/grafana-agent`:

```file=/etc/grafana-agent.yaml
...
  redis_exporter:
    enabled: true
    relabel_configs:
      - replacement: <hostname>
        target_label: instance
    redis_addr: "redis://localhost:6379"
    redis_password: $REDIS_PASSWORD
...
```

Example config using a password file:

```file=/etc/grafana-agent.yaml
...
  redis_exporter:
    enabled: true
    relabel_configs:
      - replacement: <hostname>
        target_label: instance
    redis_addr: "redis://localhost:6379"
    redis_password_file: /etc/grafana-agent/redis.json
...
```

We have tried both a simple file containing just the redis password, and the upstream `redis_exporter` formats for the password file:

```file=/etc/grafana-agent/redis.json
<password>
```
and
```file=/etc/grafana-agent/redis.json
{"redis://localhost:6379": "<password>"}
```

**Comment by spartan0x117** [comment]
Hello! I'm taking a look into this, could you try ensuring there is no newline at the end of the simple password file format?

You should be able to write a file with no newline with: `echo -n "password" >> /path/to/file`

**Comment by jesseops** [comment]
🤦🏼‍♂️ waddya know, I wasn't stripping newlines when creating the basic password file. I can confirm that stripping the newline does result in being able to authenticate successfully.

Beyond that, it looks like you've gone ahead and implemented the upstream config loader 💪🏼 looking forward to that, but for now I'm able to authenticate successfully using the basic passwordfile.

**Comment by rfratto** [comment]
Closed by #2484 and #2490. Thanks for reporting!
